Packages/com.unity.inputsystem/InputSystem/Actions/InputAction.cs

```diff
@@ -1791,7 +1791,80 @@ internal int BindingIndexOnMapToBindingIndexOnAction(int indexOfBindingOnMap)
         /// Information provided to action callbacks about what triggered an action.
         /// </summary>
         /// <remarks>
+        /// The callback context represents the current state of an <see cref="action"/> associated with the callback
+        /// and provides information associated with the bound <see cref="control"/>, its value as well as its
+        /// <see cref="phase"/>.
+        ///
+        /// The callback context provides means to consume events (push-based input) as part of an update when using
+        /// input action callback notifications, e.g. <see cref="InputAction.started"/>,
+        /// <see cref="InputAction.performed"/>, <see cref="InputAction.canceled"/> rather than relying on
+        /// pull-based reading.
```

Comment:
Is there a piece of documentation to link to (push/pull based reading) to clarify this statement?

Reply:
It's just referencing the general concept of push vs pull architecture, i.e. callbacks vs polling in this case. Would it be better to switch to those?

Reply:
I was just wondering if there is a documentation about the push/pull architecture, differences and advantages in the InputSystem, then this would be a good place to link it, but if there isn't it is fine I think.

Reply:
Thanks, I added the link as an hyperlink.